utils/wasm-gen/src/generator/syscalls/imports.rs

```diff
@@ -57,9 +57,16 @@ pub struct SysCallsImportsGeneratorInstantiator<'a, 'b>(
     ),
 );
 
+/// The set of sys-calls that need to be imported to create precise sys-call.
+#[derive(thiserror::Error, Debug)]
+#[error("The following sys-calls must be imported: {0:?}")]
+pub struct RequiredSysCalls(&'static [SysCallName]);
```

Comment:
Move it below `PreciseSysCallError` please

Reply:
plain type definition is ok to be on the top